### DISCUSSION TITLE
`out` parameter with `InterpolatedStringHandler`

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#7341] [General] [answered]
Hi,
I saw that [gist](https://gist.github.com/akarpov89/03c53ea537451c7790341d7f697a89ec) some time ago a bout a C style of parsing string based on `InterpolatedStringHandler`,
And I tried to use `out` parameter instead of `ref` one, to have a nicer way of defining values.
But the compiler tells me that I try to use unassigned local values.

How can I tell the compiler that the `out` values (`name` and `age`) are fully assigned ?

```C#
// TryParse below use a custom Interpolated String Handler
if ("Name: FaustVX Age: 32".TryParse($"Name: {Get(out string? name)} Age: {Get(out int age)}"))
    Console.WriteLine($"{name}, {age}"); // <-- error CS0165: Use of unassigned local variable 'name' and 'age'
```

Here is a shortened version on [sharplab.io](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQCUCuAdgC4CWAtgKbYDCA9uQA6kA2lATgMocBupAxpQDOAbmRpMcAGzoACgEN2Q0oQDmAUQAexSoWV09YpMlIAzdAAoARADl5VEOgBi8/EOIA1ABroAgqspHAGYYK2wAFXYATwUlSgsAElt7QPQAbwBxSmILOnxiSVQAfnRCFIBKAF8/AMdM7Nz89BUC+QCqq3Ly5DTkdH7JAE5rMqpOowH0AHoprGGktNHKSvHpqfQAHgBaLfQOdjp2dBpOVGkAVkcAVSFKdDpzInkhZVVCSmB0Fjp+eRZ0HiKUjyABGbHQAHIlhDkJVkJQWLc+gM5tYACJ0YSlOgFcjyYj8AAW42Q4iCkhksDkimUai0Oj0pAMQh6yP6UHJWBkHj++EoG3CRQAfOgssQBULGgVBehFKpukhJgBeEXvADuUtl7HlRjZmE50nQILodH+kRiNPixEJpCEhWahEY+QANOgANoASRIHEYpvxH04xHYKlUAAl5IRgGx2L5tfgqCQLEsHhYVE7iF0ALrodiUczm2K3cOR6PoQkRqMcBXKkVBvlGOHGFDk3PyYAGFhRdBEITyUx3XPmdzsfD8Ao8lh8iWsxUo8kTvmamVy6sDXqzyb9K56PvUTgAa1IjC9pByeVa2vKEwGjcbyE93vYvpY/uAgeDamLlfYmbJObz6DDqO0rRIWlBftGM6TBy6AFpaEEcKmJBfKeHB/AAMroqjWq6LToKYhx4sQOjAPQRDEK6uCUG2ADyhCdpwjARhsRKKCKab5Ku/RpOgd4bvqRomv8viMIwujAOhqHsH8FhYBgLBSX8XGbiq6B1pQur8TBxqmn4oniU4hH4iRFgLvycnCugjAvoIhKmsAVZ6jWakjhppJaeSOnCfpkaGewREmWZGwtCK1nyLZ9mOfxznqQ2yBAA)

**Comment by 333fred** [comment, marked as answer]
The problem with both the gist and this simpler example is that the compiler has no way to relate the `bool` returns of the handler to the result of `TryParse`. For all it knows, one of the `AppendX` methods could have failed, causing the subsequent ones to never be run. In order to get this (and the gist) to work, you'd have to make them return `void`, not `bool`, and have the handler just no-op for subsequent calls: https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQCUCuAdgC4CWAtgKbYDCA9uQA6kA2lATgMocBupAxpQDOAbmRpMcAGzoACgEN2Q0oQDmAUQAexSoWV09YpMlIAzdAAoARADl5VEOgBi8/EOIA1ABroAgqspHAGYYK2wAFXYATwUlSgsAElt7QPQAbwBxSmILOnxiSVQAfnRCFIBKAF8/AMdM7Nz89BUC+QCqq3Ly5DTkdH7JAE5rMqpOowH0AHoprGGktNHKSvHpqfQAHgBaLfQOdjp2dBpOVGkAVkcAVSFKdDpzInkhZVVCSmB0Fjp+eRZ0HiKUjyABGbHQAHIlhDkJVkJQWLc+gM5tYACJ0YSlOgFcjyYj8AAW42Q4iCkhksDkimUai0Oj0pAMQh6yP6UHJWBkHj++EoG3CRQAfOgssQBULGgVBehFKpukhJgBeEXvADuUtl7HlRjZmE50nQILodH+kRiNPixEJpCEhWahEY+QANOgANoASRIHEYpvxH04xHYKlUAAl5IRgGx2L5tfgqCQLEsHhYVE7iF0ALrodiUczm2K3cOR6PoQkRqMcBXKkVBvlGOHGFDk3PyYAGFhRdBEITyUx3XPmdzsfD8Ao8lh8iWsxUo8kTvmamVy6sDXqzyb9K56PvUTgAa1IjC9pByeVa2vKEwGjcbyE93vYvpY/uAgeDamLlfYmbJObz6DDqO0rRIWlBftGM6TBy6AFpaEEcKmJBfKeHB/AAMroqjWq6LToKYhx4sQOjAPQRDEK6uCUG2ADyhCdpwjARhsRKKCKab5Ku/RpOgd4bvqmAACx+IwjC6MA6GoewfwWFgGAsFJfzlOkfHQZywm+KJ4lOIR+IkRYC78nJwroIwL6CISprAFWKmkvxMFQBpWmRjp7BEfphkbC0IpmfIFlWTZaS3sgQA===.

**Reply by FaustVX** [reply]
Ok, thanks,
I know the constructor of the Handler can have an `out bool` parameter, but it won't help for my situation.
It could be nice to have some kind of `bool` returning `Closing` method on the handler to validate if everything works well, but I don't know if it could be useful in other use cases.
